lora/models.py

```diff
@@ -25,6 +24,8 @@ class LoRALinear(nn.Module):
     @staticmethod
     def from_linear(linear: nn.Linear, rank: int = 8):
         output_dims, input_dims = linear.weight.shape
+        if isinstance(linear, nn.QuantizedLinear):
+            input_dims *= 2 * linear.bits
```

Comment:
That should be `input_dims *= 32 // linear.bits` so that it works with bits other than 4. But obviously this points towards adding the following properties in `nn.QuantizedLinear` and `nn.Linear`:
- input_dims
- output_dims
- dtype

Reply:
Maybe we should add a TODO to simplify those when/if the above properties are added in a future release.

Reply:
We should definitely add those properties. No reason not to.. it's also easy to get the weight shapes mixed up.